src/sinks/appsignal/config.rs

```diff
@@ -0,0 +1,199 @@
+use futures::FutureExt;
+use http::{header::AUTHORIZATION, Request, Uri};
+use hyper::Body;
+use tower::ServiceBuilder;
+use vector_common::sensitive_string::SensitiveString;
+use vector_config::configurable_component;
+use vector_core::{
+    config::{proxy::ProxyConfig, AcknowledgementsConfig, DataType, Input},
+    tls::{MaybeTlsSettings, TlsEnableableConfig},
+};
+
+use crate::{
+    codecs::Transformer,
+    http::HttpClient,
+    sinks::{
+        prelude::{SinkConfig, SinkContext},
+        util::{
+            http::HttpStatusRetryLogic, BatchConfig, Compression, ServiceBuilderExt,
+            SinkBatchSettings, TowerRequestConfig,
+        },
+        BuildError, Healthcheck, HealthcheckError, VectorSink,
+    },
+};
+
+use super::{
+    service::{AppsignalResponse, AppsignalService},
+    sink::AppsignalSink,
+};
+
+/// Configuration for the `appsignal` sink.
+#[configurable_component(sink("appsignal", "Deliver log and metric event data to AppSignal."))]
+#[derive(Clone, Debug, Default)]
+pub struct AppsignalConfig {
+    /// The URI for the AppSignal API to send data to.
+    #[configurable(validation(format = "uri"))]
+    #[configurable(metadata(docs::examples = "https://appsignal-endpoint.net"))]
+    #[serde(default = "default_endpoint")]
+    pub endpoint: String,
+
+    /// A valid app-level AppSignal Push API key.
+    #[configurable(metadata(docs::examples = "00000000-0000-0000-0000-000000000000"))]
+    #[configurable(metadata(docs::examples = "${APPSIGNAL_PUSH_API_KEY}"))]
+    push_api_key: SensitiveString,
+
+    #[configurable(derived)]
+    #[serde(default = "Compression::gzip_default")]
+    compression: Compression,
+
+    #[configurable(derived)]
+    #[serde(default)]
+    batch: BatchConfig<AppsignalDefaultBatchSettings>,
+
+    #[configurable(derived)]
+    #[serde(default)]
+    request: TowerRequestConfig,
+
+    #[configurable(derived)]
+    tls: Option<TlsEnableableConfig>,
+
+    #[configurable(derived)]
+    #[serde(
+        default,
+        skip_serializing_if = "crate::serde::skip_serializing_if_default"
+    )]
+    encoding: Transformer,
+
+    #[configurable(derived)]
+    #[serde(
+        default,
+        deserialize_with = "crate::serde::bool_or_struct",
+        skip_serializing_if = "crate::serde::skip_serializing_if_default"
+    )]
+    acknowledgements: AcknowledgementsConfig,
+}
+
+pub(crate) fn default_endpoint() -> String {
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏

I believe all of the `pub(crate)` in the `appsignal/` directory can be made `pub(super)`

```suggestion
pub(super) fn default_endpoint() -> String {
```